### ISSUE TITLE
Slow check for similar generated files that differ by case

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8363]
In jOOQ 3.10, we've introduced a check that verifies if there is any previously generated file that has the same name but different case (see #5892). This can be a problem when e.g. renaming a table from `TABLE_NAME` to `TABLENAME` and regenerating the code. On Unix systems, this would produce two distinct file names. Not so on Windows.

The currently implemented check is very slow:

```java
if (file.exists()) {

    // [#5892] On Windows FAT or NTFS and other case-insensitive file systems, we must
    //         explicitly replace files whose case-sensitive file name has changed
    String[] list = file.getParentFile().list(new FilenameFilter() {
        @Override
        public boolean accept(File dir, String name) {
            return name.equalsIgnoreCase(file.getName()) && !name.equals(file.getName());
        }
    });

    if (list != null && list.length > 0)
        file.delete();
```

A profiling session shows 50% of time is spent in this `list()` call for a larger code generation run using a `DDLDatabase` (so connecting to the dictionary views is not a bottleneck here):

![image](https://user-images.githubusercontent.com/734593/53566026-65f30980-3b5b-11e9-98e3-02285188b1d0.png)

We should probably cache the files for any given parent file and apply the `FilenameFilter` on that cache rather than on the file every time, as the files are all in the same directory.

**Comment by lukaseder** [comment]
The fixed version shows that a lot less time is spent in this logic:

![image](https://user-images.githubusercontent.com/734593/53568464-5f1bc500-3b62-11e9-9fc8-a6df22301bbf.png)

In fact, we can now see that the creation of the directory takes as much time as listing existing files:

![image](https://user-images.githubusercontent.com/734593/53568502-79ee3980-3b62-11e9-8764-3347b19fcdde.png)

That's another thing we can cache with the newly introduced cache.

**Comment by lukaseder** [comment]
The `mkdirs()` call disappears from the profiling session (obviously)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.12. Backport scheduled for 3.11.10 (#8364) and 3.10.9 (#8365)
